dalek-cryptography/curve25519-dalek
# Rustfmt

Ok so there's a lot going on here:

1. I undid some bad formatting changes from a previous commit
2. I added `#[rustfmt::skip]` in a bunch of places. Thanks to #420 for that!
3. I did some manual (whitespace only) formatting of the functions in step 2
4. I ran `cargo fmt`

Here is how to verify each step:

* Install [`difftastic`](https://github.com/Wilfred/difftastic/). It's a syntactic differ that will make your life much much easier.
* Check step 1 by running `GIT_EXTERNAL_DIFF=difft git show a32f56b --ext-diff`
* Check step 2 by visually checking commits `6d906bb` and `a959787`. They're intentionally small.
* Check step 3 by running `GIT_EXTERNAL_DIFF=difft git show 95b368a --ext-diff`.
* Check step 4 by doing the following:
    * Hash the relevant commit's diff (using the ordinary difftool): `git diff d2bf310^..d2bf310 | shasum -a 256`
    * Check out the immediate ancestor: `git checkout a959787`
    * Run `cargo fmt`
    * Now hash the diff and compare to the previous diff: `git diff | shasum -a 256`
    * If the two hashes are the same, then this is indeed a `cargo fmt`

I've attached screenshots for each of these checks. My `cargo --version` is 1.64.0.

Check step 1:
<img width="889" alt="reformatting_check" src="https://user-images.githubusercontent.com/752802/198735286-57098c5c-19c6-4ffc-bf76-ede1160329ca.png">

Check step 2
<img width="889" alt="whitespace_check" src="https://user-images.githubusercontent.com/752802/198735288-edf6161c-53b1-444d-a639-69e917dfb3cb.png">

Check step 4
<img width="781" alt="cargo_fmt_check" src="https://user-images.githubusercontent.com/752802/198735282-1fe700a5-ced7-4e43-9fd7-ee22682ce3e1.png">
